cli/cli.go

```diff
@@ -112,48 +132,189 @@ func newCli(
 			password := addCmd.StringOpt("p password", "", "Password")
 
 			addCmd.Action = func() {
-				exitWith("", core.Auth().Add(context.TODO(), *resources, *username, *password))
+				node, err := nodeProvider.CreateNodeIfNotExists()
```

Comment:
Here's what I mean https://github.com/opctl/opctl/pull/870

Reply:
I agree with having each top level command in it's own file being a nice end state. The reason I didn't do it in this PR was this is already a multi-day PR that touches alot of common areas so I wanted to get this in to main and then fast follow with the small change of breaking the functions out. This PR is already a positive change without that further enhancement IMHO so I wanted to get it in. The main topic of this PR was to get rid of core and the NodeHandle interface as they just introduced LOC/abstractions(i.e. complexity) with arguably no/insufficient value add.

Reply:
Not related directly but I also want to hopefully fast follow by looking at using http://onsi.github.io/gomega/#gexec-testing-external-processes for our CLI tests. I'm hopeful it could allow us to have end to end tests over each CLI command. Right now our CLI tests don't really assert anything useful.